### PR TITLE
feat: adding func to insert user into DB

### PR DESCRIPTION
**Pull request by lifehou5e**

### CODE REVIEW

File: app/repository/pg/user.go [hunk 1]
```diff
@@ -22,6 +23,14 @@ func NewRepo(opt cfg.Options) *Repo {
...
 // and will make our app flexible and maintainable.
 func (r *Repo) Add(usr entities.User) error {
-	// INSERT INTO "users" ...
+	sqlStatement := `
+INSERT INTO users (first_name, last_name, email, password)
+VALUES ($1, $2, $3, $4)
+RETURNING id`
+	err := r.QueryRow(sqlStatement, usr.FirstName, usr.LastName, usr.Email, usr.Password)
+	if err != nil {
+		log.Println(err)
```
[the remark this answers is not in the document]
Comment (Rusli4k) [reply app/repository/pg/user.go:32]: and change "usr" with "user"

File: app/repository/pg/user.go [hunk 2]
```diff
@@ -22,6 +23,14 @@ func NewRepo(opt cfg.Options) *Repo {
 // without even knowing it that allow us to decouple our layers
 // and will make our app flexible and maintainable.
 func (r *Repo) Add(usr entities.User) error {
-	// INSERT INTO "users" ...
+	sqlStatement := `
+INSERT INTO users (first_name, last_name, email, password)
+VALUES ($1, $2, $3, $4)
+RETURNING id`
```
Comment (delveper) [inline app/repository/pg/user.go:29]: I do not get why we have `RETURNING id` here

Comment (Brigant) [inline app/repository/pg/user.go]: I think that if our function does not return anything but an error, then the `RETURNING id`  in the request is not necessary

File: app/repository/pg/user.go [hunk 3]
```diff
@@ -22,6 +23,14 @@ func NewRepo(opt cfg.Options) *Repo {
...
+	err := r.QueryRow(sqlStatement, usr.FirstName, usr.LastName, usr.Email, usr.Password)
```
[the remark this answers is not in the document]
Comment (lifehou5e) [reply app/repository/pg/user.go:30]: Sure thing.

File: app/repository/pg/user.go (hunk 2 above)
Comment (lifehou5e) [reply app/repository/pg/user.go:29]: I'll remove it.

File: app/repository/pg/user.go [hunk 4]
```diff
@@ -22,6 +23,14 @@ func NewRepo(opt cfg.Options) *Repo {
 // without even knowing it that allow us to decouple our layers
 // and will make our app flexible and maintainable.
 func (r *Repo) Add(usr entities.User) error {
-	// INSERT INTO "users" ...
+	sqlStatement := `
```
[the remark this answers is not in the document]
Comment (lifehou5e) [reply app/repository/pg/user.go]: Yeah, seems it won't and shouldn't change, makes sense. TY. 

fixed

File: app/repository/pg/user.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lifehou5e) [reply app/repository/pg/user.go]: fixed

File: app/repository/pg/user.go (hunk 2 above)
Comment (lifehou5e) [reply app/repository/pg/user.go]: There was the same comment from @Brigant , fixed that in commit above.

**Review by Brigant — DISMISSED** [review]

**Review by Rusli4k — DISMISSED** [review]

**Review by delveper — DISMISSED** [review]

File: app/repository/pg/user.go [hunk 5]
```diff
@@ -21,7 +21,15 @@ func NewRepo(opt cfg.Options) *Repo {
 // Add meth implements usecase.UserRepository interface
 // without even knowing it that allow us to decouple our layers
 // and will make our app flexible and maintainable.
-func (r *Repo) Add(usr entities.User) error {
-	// INSERT INTO "users" ...
+func (r *Repo) Add(user entities.User) error {
+	const sqlStatement = `INSERT INTO "user" (first_name, last_name, email, password)
+	VALUES ($1, $2, $3, $4)`
+
+	_, err := r.Exec(sqlStatement, user.FirstName, user.LastName, user.Email,
```
Comment (vpakhuchyi) [inline app/repository/pg/user.go:28]:
```
if _, err := r.Exec(sqlStatement, user.FirstName, user.LastName, user.Email, user.Password); err != nil {
	return err
}
```
Looks better.

Comment (lifehou5e) [reply app/repository/pg/user.go]: fixed, TY

**Review by vpakhuchyi — APPROVED** [review]

**Review by vdshakh — APPROVED** [review]

**Review by Brigant — APPROVED** [review]

**Review by Rusli4k — APPROVED** [review]

**Review by delveper — APPROVED** [review]

**Pull request merged**